File: 프로그래머스/기초 트레이닝 - Level 0/연속 부분 수열 합의 개수-ex.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        Set<Integer> set = new HashSet<>();
        int sum; 
        
        for (int length = 1; length <= elements.length; length++) {
            for (int start = 0; start < elements.length; start++) {
                sum = 0; 
                for(int k = start; k < start + length; k++) {
                    sum += elements[k % elements.length];
                }
                set.add(sum);
            }
        }
        
        return set.size();
    }
}

File: 프로그래머스/기초 트레이닝 - Level 0/연속 부분 수열 합의 개수-ex.py
def solution(elements):
    length = len(elements)
    seq_sum_set = set()
    
    for i in range(length):
        seq_sum = elements[i]
        seq_sum_set.add(seq_sum)
        
        for j in range(i + 1, i + length):
            seq_sum += elements[j % length]
            seq_sum_set.add(seq_sum)
    
    return len(seq_sum_set)